code/modules/crafting/table.dm

```diff
@@ -121,11 +121,13 @@
 							amt -= RC.reagents.get_reagent_amount(RG.id)
 							RC.reagents.del_reagent(RG.id)
 
-	for(var/A in R.parts)
+	deletion_loop:
 		for(var/B in Deletion)
-			if(!istype(B, A))
-				Deletion.Remove(B)
-				qdel(B)
+			for(var/A in R.parts)
+				if(istype(B, A))
+					continue deletion_loop
+			Deletion.Remove(B)
+			qdel(B)
```

Comment:
You made the code do the same thing in 6 lines instead of 4 lines
I dunno if win in less 1 boolean negation per cycle is worth it